### PR TITLE
Catalog Server Accepts Compressed Updates

### PR DESCRIPTION
**Pull request by dthain**
Partially addresses #1732. @nhazekam you will need to test with your half of the solution.

### CODE REVIEW

**Comment by dthain** [comment]
Go ahead and test them together, then merge when you are satisfied.

File: dttools/src/catalog_server.c [hunk 1]
```diff
@@ -216,19 +223,37 @@ static void make_hash_key(struct jx *j, char *key)
...
 	char key[LINE_MAX];
 	int port;
-	int result;
+	unsigned long data_length, raw_data_length;
 	struct jx *j;
 
 	while(1) {
-		result = datagram_recv(update_port, data, DATAGRAM_PAYLOAD_MAX, addr, &port, 0);
-		if(result <= 0)
+		raw_data_length = datagram_recv(update_port, raw_data, DATAGRAM_PAYLOAD_MAX, addr, &port, 0);
```
Comment (nhazekam) [inline dttools/src/catalog_server.c]: `datagram_recv` returns -1 if there is no data.

Since `raw_data_length` is unsigned it never terminates the loop.

**Comment by dthain** [comment]
Quite right, how about this?

**Comment by nhazekam** [comment]
That change works great.

**Pull request merged**